app/controllers/Checkout.scala

```diff
@@ -417,7 +417,11 @@ class Checkout(fBackendFactory: TouchpointBackends, commonActions: CommonActions
       // should a client want not to proxy via this server.
       Ok(Json.obj("Addresses" -> result.Addresses))
     } recover {
-      case e => BadRequest(Json.obj("Message" -> e.getMessage))
+      case error if error.getMessage == "Bad Request" =>
+        BadRequest //The postcode was invalid
```

Comment:
The client is not using the error response: https://github.com/guardian/subscriptions-frontend/blob/e057a5b8dfd8ec94c48ebae4455f0e1fcb6aed5c/assets/javascripts/modules/checkout/addressFinder.jsx#L42-L49